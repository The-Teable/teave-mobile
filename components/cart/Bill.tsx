import styled from "styled-components";
import CenteredContainer from "../layout/CenteredContainer";

const Bill = () => {
  return (
    <>
      <CenteredContainer>
        <S.Container>
          <S.Wrapper>
            <S.ItemGray>총 상품 금액</S.ItemGray>
            <S.Item>67,000원</S.Item>
          </S.Wrapper>
          <S.Wrapper>
            <S.ItemGray>총 배송비</S.ItemGray>
            <S.Item>3,000원</S.Item>
          </S.Wrapper>
          <S.Wrapper>
            <S.ItemGray>총 할인 금액</S.ItemGray>
            <S.Item>0원</S.Item>
          </S.Wrapper>
          <S.Wrapper>
            <S.Result>결제 금액</S.Result>
            <S.Result>70,000원</S.Result>
          </S.Wrapper>
        </S.Container>
      </CenteredContainer>
    </>
  );
};

export default Bill;

const S: any = {};

S.Container = styled.div`
  display: flex;
  flex-direction: column;
  background-color: #ffffff;
  padding: 2rem 1.5rem;
`;
S.Wrapper = styled.div`
  display: flex;
  justify-content: space-between;
  margin: 1rem;
`;
S.ItemGray = styled.div`
  color: #808080;
  font-size: 1.4rem;
`;
S.Item = styled.div`
  font-size: 1.4rem;
  font-weight: 500;
`;
S.Result = styled.div`
  font-size: 1.4rem;
  font-weight: bold;
`;
