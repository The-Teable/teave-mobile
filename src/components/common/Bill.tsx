import styled from "styled-components";
// import CenteredContainer from "../layout/CenteredContainer";

const Bill = (props: {
  productPrice: number;
  deliveryCost: number;
  discountPrice: number;
}) => {
  const { productPrice, deliveryCost, discountPrice } = props;
  return (
    <>
      <S.Container>
        <S.Wrapper>
          <S.ItemGray>총 상품 금액</S.ItemGray>
          <S.Item>{productPrice.toLocaleString() + "원"}</S.Item>
        </S.Wrapper>
        <S.Wrapper>
          <S.ItemGray>총 배송비</S.ItemGray>
          <S.Item>{deliveryCost.toLocaleString() + "원"}</S.Item>
        </S.Wrapper>
        <S.Wrapper>
          <S.ItemGray>총 할인 금액</S.ItemGray>
          <S.Item>{discountPrice.toLocaleString() + "원"}</S.Item>
        </S.Wrapper>
        <S.Wrapper>
          <S.Result>결제 금액</S.Result>
          <S.Result>
            {(productPrice + deliveryCost - discountPrice).toLocaleString() +
              "원"}
          </S.Result>
        </S.Wrapper>
      </S.Container>
    </>
  );
};

export default Bill;

const S: any = {};

S.Container = styled.div`
  display: flex;
  flex-direction: column;
  background-color: #ffffff;
  padding: 1rem 1rem;
`;
S.Wrapper = styled.div`
  display: flex;
  justify-content: space-between;
  padding: 1rem 0;
  &:last-child {
    border-top: 1px #f4f4f4 solid;
  }
`;
S.ItemGray = styled.div`
  color: #808080;
  font-size: 1.4rem;
`;
S.Item = styled.div`
  font-size: 1.4rem;
  font-weight: 500;
`;
S.Result = styled.div`
  font-size: 1.6rem;
  font-weight: bold;
`;
