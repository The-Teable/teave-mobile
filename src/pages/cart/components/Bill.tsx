import styled from "styled-components";
import CenteredContainer from "../../../components/layout/CenteredContainer";

const Bill = () => {
  return (
    <>
      <CenteredContainer>
        <S.Container>
          <S.Wrapper>
            <S.ItemGray>총 상품 금액</S.ItemGray>
            <S.Item>67,000원</S.Item>
          </S.Wrapper>
          <S.Wrapper>
            <S.ItemGray>총 배송비</S.ItemGray>
            <S.Item>3,000원</S.Item>
          </S.Wrapper>
          <S.Wrapper>
            <S.ItemGray>총 할인 금액</S.ItemGray>
            <S.Item>0원</S.Item>
          </S.Wrapper>
          <S.Wrapper>
            <S.Result>결제 금액</S.Result>
            <S.Result>70,000원</S.Result>
          </S.Wrapper>
        </S.Container>
      </CenteredContainer>
    </>
  );
};

export default Bill;

const S: any = {};

S.Container = styled.div`
  display: flex;
  flex-direction: column;
  background-color: #ffffff;
  padding: 1rem 1rem;
`;
S.Wrapper = styled.div`
  display: flex;
  justify-content: space-between;
  padding: 1rem 0;
  &:last-child {
    border-top: 1px #f4f4f4 solid;
  }
`;
S.ItemGray = styled.div`
  color: #808080;
  font-size: 1.4rem;
`;
S.Item = styled.div`
  font-size: 1.4rem;
  font-weight: 500;
`;
S.Result = styled.div`
  font-size: 1.6rem;
  font-weight: bold;
`;
