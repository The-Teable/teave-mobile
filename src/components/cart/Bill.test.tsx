import Bill from "./Bill";
import { render } from "@testing-library/react";

describe("Bill components", () => {
    it("matches snapshot", () => {
        const utils = render(<Bill productPrice={0} deliveryCost={0} discountPrice={0}/>);
        expect(utils.container).toMatchSnapshot();
    })

    it("shows the props correctly", () => {
        const utils = render(<Bill productPrice={12000} deliveryCost={3000} discountPrice={1500}/>);
        utils.getByText("총 상품 금액");
        utils.getByText("12,000원");
        utils.getByText("총 배송비");
        utils.getByText("3,000원");
        utils.getByText("총 할인 금액");
        utils.getByText("1,500원");
        utils.getByText("결제 금액");
        utils.getByText("13,500원");
    })
})
