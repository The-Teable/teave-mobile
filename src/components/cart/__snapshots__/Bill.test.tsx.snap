// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bill components matches snapshot 1`] = `
<div>
  <div
    class="CenteredContainer-sc-1yuzjpv-0 fqZIID"
  >
    <div
      class="Bill__Container-sc-gazs1n-0 MqvCF"
    >
      <div
        class="Bill__Wrapper-sc-gazs1n-1 cjHyiK"
      >
        <div
          class="Bill__ItemGray-sc-gazs1n-2 dQLWtA"
        >
          총 상품 금액
        </div>
        <div
          class="Bill__Item-sc-gazs1n-3 curGih"
        >
          0원
        </div>
      </div>
      <div
        class="Bill__Wrapper-sc-gazs1n-1 cjHyiK"
      >
        <div
          class="Bill__ItemGray-sc-gazs1n-2 dQLWtA"
        >
          총 배송비
        </div>
        <div
          class="Bill__Item-sc-gazs1n-3 curGih"
        >
          0원
        </div>
      </div>
      <div
        class="Bill__Wrapper-sc-gazs1n-1 cjHyiK"
      >
        <div
          class="Bill__ItemGray-sc-gazs1n-2 dQLWtA"
        >
          총 할인 금액
        </div>
        <div
          class="Bill__Item-sc-gazs1n-3 curGih"
        >
          0원
        </div>
      </div>
      <div
        class="Bill__Wrapper-sc-gazs1n-1 cjHyiK"
      >
        <div
          class="Bill__Result-sc-gazs1n-4 iQoUGo"
        >
          결제 금액
        </div>
        <div
          class="Bill__Result-sc-gazs1n-4 iQoUGo"
        >
          0원
        </div>
      </div>
    </div>
  </div>
</div>
`;
